disnake/guild.py

```diff
@@ -4191,3 +4204,464 @@ async def timeout(
+                "ThreadArchiveDurationLiteral", try_enum_to_int(default_auto_archive_duration)
+            )
+
+        return _id
+
+    def add_voice_channel(
+        self,
+        name: str,
+        *,
+        overwrites: Dict[PlaceholderID, PermissionOverwrite] = MISSING,
+        category: PlaceholderID = MISSING,
+        slowmode_delay: int = MISSING,
+        nsfw: bool = MISSING,
+        bitrate: int = MISSING,
+        user_limit: int = MISSING,
+        rtc_region: Optional[Union[str, VoiceRegion]] = MISSING,
+        video_quality_mode: VideoQualityMode = MISSING,
+    ) -> PlaceholderID:
+        """
+        Adds a voice channel to the guild builder.
+
+        Parameters
+        ----------
+        name: :class:`str`
+            The channel's name.
+        overwrites: Dict[``PlaceholderID``, :class:`PermissionOverwrite`]
+            A :class:`dict` of roles to :class:`PermissionOverwrite`\\s to apply to the channel.
+        category: ``PlaceholderID``
+            The category to place the new channel under.
+
+            .. warning::
+                Unlike :func:`Guild.create_voice_channel`, the parent category's
+                permissions will *not* be synced to this new channel by default.
+
+        slowmode_delay: :class:`int`
+            Specifies the slowmode rate limit for users in this channel, in seconds.
+            A value of ``0`` disables slowmode. The maximum value possible is ``21600``.
+            If not provided, slowmode is disabled.
+        nsfw: :class:`bool`
+            Whether to mark the channel as NSFW or not.
+        bitrate: :class:`int`
+            The channel's preferred audio bitrate in bits per second.
+        user_limit: :class:`int`
+            The channel's limit for number of members that can be in a voice channel.
+        rtc_region: Optional[Union[:class:`str`, :class:`VoiceRegion`]]
+            The region for the voice channel's voice communication.
+            A value of ``None`` indicates automatic voice region detection.
+        video_quality_mode: :class:`VideoQualityMode`
+            The camera video quality for the voice channel's participants.
+
+        Returns
+        -------
+        ``PlaceholderID``
+            A placeholder ID for the created voice channel.
+        """
+        _id, data = self._add_channel(
+            type=ChannelType.voice,
+            name=name,
+            overwrites=overwrites,
+            category=category,
+            slowmode_delay=slowmode_delay,
+            nsfw=nsfw,
+        )
+
+        if bitrate is not MISSING:
+            data["bitrate"] = bitrate
+
+        if user_limit is not MISSING:
+            data["user_limit"] = user_limit
+
+        if rtc_region is not MISSING:
+            data["rtc_region"] = None if rtc_region is None else str(rtc_region)
+
+        if video_quality_mode is not MISSING:
+            data["video_quality_mode"] = try_enum_to_int(video_quality_mode)
+
+        return _id
```

Comment:
Any plans for an `add_stage_channel` function?

Reply:
I don't think they can be created. Stage channels are only for community guilds and as far as I can tell a guild created cannot be set to COMMUNITY on creation.

Reply:
Ah good point

Reply:
Yup, tried that as well. It doesn't look like you can set features on creation, and the only allowed channel types are 0 (text), 2 (voice), and 4 (category).